src/main/resources/jenkins/plugins/office365connector/Webhook/global.jelly

```diff
@@ -7,5 +7,25 @@
             <f:entry title="Name" field="globalName">
                 <f:textbox/>
             </f:entry>
+        <f:advanced>
+            <f:section title="Proxy Config">
+                <f:description>
+                    This section defines the Office365Connector proxy settings that will apply to all webhook notifications. This
+                    proxy configuration will be prioritized over the Jenkins global proxy config
```

Comment:
for what cases we need overwrite global settings?

Reply:
In the case where the Jenkins global proxy allows all traffic to go enabling the plugin to function but other areas of the environment are negatively affected. 

Example: I setup the global proxy and the plugin worked however other functionality like being able to download other files to the jenkins agents failed. It was investigated and found that the proxy server blocked that feature. A way around it was to setup the 'No Proxy Host', but i have over a 100 servers and that was deemed to major of a risk to configure and test especially with production systems.

So having the proxy config local to the plugin means the plugin can make use of a proxy just for its specific purpose without affecting the rest of Jenkins, if a proxy is not configured via the plugin, then it will make use of the global proxy if configured.

Reply:
OK, still waiting for fixing other issues requested by code review

Reply:
Hi, chiming in as I'm also interested in getting proxy feature working well in this plugin.

1. Are you sure that local proxy config will overwrite global config? Looking at flow of method `getHttpClient`, local proxy config:
https://github.com/jenkinsci/office-365-connector-plugin/blob/09bac61fcb495bea96b9d6a6b74a846710db7730/src/main/java/jenkins/plugins/office365connector/HttpWorker.java#L111
Will always be overwritten by global proxy config, if present:
https://github.com/jenkinsci/office-365-connector-plugin/blob/09bac61fcb495bea96b9d6a6b74a846710db7730/src/main/java/jenkins/plugins/office365connector/HttpWorker.java#L123

2. It'd be clearer to use only one class regarding proxy settings. Existing class `hudson.ProxyConfiguration` represents similar data as your class `jenkins.plugins.office365connector.Proxy` and I presume it could be used for this purpose instead.

Reply:
Hi

The plugin proxy wont overwrite the global proxy config, it would favor the plugin proxy over the global proxy. Regarding point 2, thanks ill investigate and see if i can better the implementation.

Reply:
> The plugin proxy wont overwrite the global proxy config, it would favor the plugin proxy over the global proxy.

That's what I mean, current implementation favours global proxy. To change it how you meant, you could e.g. refactor this `if` clause to `else if` clause:
https://github.com/jenkinsci/office-365-connector-plugin/blob/09bac61fcb495bea96b9d6a6b74a846710db7730/src/main/java/jenkins/plugins/office365connector/HttpWorker.java#L117

Reply:
Refactored this so that priority is plugin proxy, then Jenkins global proxy